### DISCUSSION TITLE
Metronome visuals

### DISCUSSION THREAD

**Discussion by BigHonkinDaddy** [powertab/powertabeditor#465] [Ideas]
The audio metronome available with PowerTab's midi player is good, but I'd like to see visual clues that metronome apps typically have.  That would help keeping count, not just the beat.

**Comment by cameronwhite** [comment]
Are there any visual styles in particular that you like from other metronome apps?

**Comment by BigHonkinDaddy** [comment]
Something like this could work.  Even just the four (given 4 beats/measure) lights at the top would be enough.

https://github.com/powertab/powertabeditor/assets/77425146/4b14c240-7a66-42ea-86f0-fb86290b2913

**Reply by BigHonkinDaddy** [reply]
Of course, the visual can be a lot smaller...something that might fit down on the status bar, something like this:
![PowerTab Metronome](https://github.com/powertab/powertabeditor/assets/77425146/335f134c-ef84-445c-897a-d83bf2e00ed8)

**Comment by cameronwhite** [comment]
Thanks!
With showing the number of beats, I think the main challenges are:
- The number can be pretty large for unusual time signatures, e.g. 13/8. I think the upper limit in the code is 32
- Since the time signature can change during the score, so need to be careful that the UI doesn't shift around too much

**Comment by BigHonkinDaddy** [comment]
Good points.  An alternative might be to have a separate window pop-up for the metronome during playback -probably not worth the effort.


-------- Original message --------
From: Cameron White ***@***.***>
Date: 2/9/24 10:48 PM (GMT-05:00)
To: powertab/powertabeditor ***@***.***>
Cc: BigHonkinDaddy ***@***.***>, Author ***@***.***>
Subject: Re: [powertab/powertabeditor] Metronome visuals (Discussion #465)


Thanks!
With showing the number of beats, I think the main challenges are:

  *   The number can be pretty large for unusual time signatures, e.g. 13/8. I think the upper limit in the code is 32
  *   Since the time signature can change during the score, so need to be careful that the UI doesn't shift around too much

**Comment by chkrvrty** [comment]
A simpler alternative would be to just have a single dot in the status bar, which flashes red for the first beat, and yellow for all the other beats. That way it doesn't matter if the time signature changes during the score, or how many beats/measure, it won't affect the visual.

**Reply by BigHonkinDaddy** [reply]
or even a single dot with a number that indicates which beat in the measure...
